### PR TITLE
Added Some Search side effects

### PR DESCRIPTION
**Pull request by justuskw**
https://trello.com/c/m3uc0ra0/33-implement-go-back-to-all-reviews-button
https://trello.com/c/rhFLPZCq/34-give-message-of-how-many-reviews-match-the-search
https://trello.com/c/UQrGubC7/35-add-clear-searchbox-button-inside-search-box

Thanks for the review!

### CODE REVIEW

File: client/src/components/app.jsx [hunk 1]
```diff
@@ -3,6 +3,7 @@ import axios from 'axios';
 import ReviewList from './reviewList';
 import Ratings from './ratings';
 import Search from './search';
+import BackToAllReviews from './backToAllReviews';
 
 export default class App extends React.Component {
```
Comment (briansai) [inline client/src/components/app.jsx:8]: Can you export default like this?  I'm not sure, but maybe put the export default at the bottom for consistency.

File: client/src/components/backToAllReviews.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+const BackToAllReviews = props => (
```
Comment (briansai) [inline client/src/components/backToAllReviews.jsx]: Include parentheses around props?

File: client/src/components/backToAllReviews.jsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  <div>hello</div>
```
Comment (briansai) [inline client/src/components/backToAllReviews.jsx]: Assuming this will be replaced with further content later?

**Review by briansai — CHANGES_REQUESTED** [review]
I made some assumptions on some parts, so please correct me if I am wrong, so I can approve it.  Thanks!

File: client/src/components/app.jsx (hunk 1 above)
Comment (justuskw) [reply client/src/components/app.jsx:8]: Yeah you can for classes. I like to do it at the top for classes and at the bottom for stateless functions.

File: client/src/components/backToAllReviews.jsx (hunk 2 above)
Comment (justuskw) [reply client/src/components/backToAllReviews.jsx]: I did at first and then eslint told be not too. It seems inconsistent when it wants me to or not, but I'm just following what it tells me. But good eye.

File: client/src/components/app.jsx (hunk 1 above)
Comment (briansai) [reply client/src/components/app.jsx:8]: Okay, cool.  Thanks for letting me know.

File: client/src/components/backToAllReviews.jsx (hunk 2 above)
Comment (briansai) [reply client/src/components/backToAllReviews.jsx]: That's interesting.  Okay cool.

File: client/src/components/backToAllReviews.jsx (hunk 3 above)
Comment (justuskw) [reply client/src/components/backToAllReviews.jsx]:
```import React from 'react';
import PropTypes from 'prop-types';

const BackToAllReviews = ({ handleBacktoAllReviewsClick, currentReviewsLength, searchedWord }) => (
  <div>
    <div className="mentions">{currentReviewsLength} guests have mentioned "{searchedWord}"</div>
    <button className="backToAllReviews" onClick={handleBacktoAllReviewsClick}>Back to all reviews</button>
  </div>
);

export default BackToAllReviews;

BackToAllReviews.propTypes = {
  handleBacktoAllReviewsClick: PropTypes.func.isRequired,
  currentReviewsLength: PropTypes.number.isRequired,
  searchedWord: PropTypes.string.isRequired,
};```

This is what that component should look like. I think you are looking at an old commit somehow.

**Review by briansai — COMMENTED** [review]
Looks great.  Keep it up!

**Review by briansai — APPROVED** [review]
Looks great!  Keep it up!

**Pull request merged**